Resources/config/events_deferring.yml

```diff
@@ -3,6 +3,11 @@ parameters:
   event_dispatcher.class: Smartbox\Integration\FrameworkBundle\Tools\EventsDeferring\EventDispatcher
 
 services:
+  event_dispatcher:
```

Comment:
To override the service directly in the yaml file, we should copy the original definition of the service and just replace what we want (in this case the class) based on symfony 3.4 documentation (https://symfony.com/doc/3.4/service_container/service_decoration.html). Checking the **event_dispatcher** definition in Symfony 3.4, the service has a tag that is not included here (see below the event_dispatcher definition)
```
        <service id="event_dispatcher" class="Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher" public="true">
            <argument type="service" id="service_container" />
            <tag name="container.hot_path" /> # Tag name
        </service>
```
This tag ```container.hot_path``` exists in Symfony 3.4 (https://symfony.com/doc/3.4/reference/dic_tags.html) but doesn't exist in Symfony 2.8
(https://symfony.com/doc/2.8/reference/dic_tags.html)
To be able have this service setup correctly in the symfony versions we support (2.8 and 3.4) I suggest to define a new compiler pass that overrides the class name of the service and leave the rest managed by the symfony version itself. This means that this service could be deleted from here, and also the parameter ```event_dispatcher.class: Smartbox\Integration\FrameworkBundle\Tools\EventsDeferring\EventDispatcher``` because this was the way to override services until Symfony 2.6 (see https://symfony.com/doc/2.6/cookbook/bundles/override.html#services-configuration)

Reply:
The current way effectively overrides the event dispatcher in a backwards compatible way without a compiler pass.

Reply:
My point here was that in Symfony 3.4 this service has the tag name "**container.hot_path**" and with the override we are missing this information. This tag doesn't exist in symfony 2.8. For this reason the point of create a compiler pass

Reply:
Seems that the tag is only used for debugging purposes: https://github.com/symfony/dependency-injection/blob/master/Dumper/PhpDumper.php#L153

I've added the tag in our definition, since it's backwards compatible with 2.8 (framework will ignore it)